### PR TITLE
Validate result and cut reason after 300 characters

### PR DESCRIPTION
**Pull request by Martchus**
This limits the length of the reason as mentioned in https://progress.opensuse.org/issues/25814#note-15. It felt a bit weird to limit the reason but allow the result to be of any length. So I also added a commit to validate the result.

### CODE REVIEW

**Comment by Martchus** [comment]
The coverage report is clearly wrong. It can not be that the single line it complains about is not covered while the other lines are covered. And this function is definitely invoked by the tests.

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 1]
```diff
@@ -1901,7 +1901,25 @@ test modules
...
+    }
+    else {
+        $result = $self->calculate_result;
+    }
+
+    # limit length of the reason
+    # note: The reason can be anything the worker picked up as useful information so better cut it at a
+    #       reasonable, human-readable length. This also avoids growing the database too big.
+    my $reason = $args{reason};
+    $reason = substr($reason, 0, 120) if defined $reason && length $reason > 120;
```
Comment (kalikiana) [inline lib/OpenQA/Schema/Result/Jobs.pm:1922]: How about adding an ellipsis here to make it visible that the string was indeed cut short?

**Comment by okurz** [comment]
@Martchus please link the coverage report about the flaky coverage to one of my "flaky" tickets so that we can cover that with consistent tests as well

**Comment by Martchus** [comment]
@okurz But the test *is* consistent. It will always call this function and the line in question is in the main branch of that function which will definitely be executed up to that point (subsequent lines are considered covered!). So unless I'm reading the results wrong it is impossible to be correct.

**Comment by Martchus** [comment]
The report would make sense when moving the green/red highlighting 2 lines to the bottom. That would mean the coverage for `$result = OBSOLETED;` is flaky.

**Comment by Martchus** [comment]
Adding elipses via the non-ASCII character "…" revealed some encoding problems. However, I now think the solution I've just pushed is not correct/neccassary and maybe `script/client` is at fault. So let's not merge this PR until this is clear.

**Comment by Martchus** [comment]
The encoding problems were really just related to the client which encoded the parameter twice. I took the opportunity to test the new client from https://github.com/os-autoinst/openQA/pull/2859 and it works. So I removed the workaround again from this PR which should be good now.

**Comment by okurz** [comment]
I merged #2925 , please rebase

**Pull request merged**